stdlib/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/io/events/records/HasNextDelimitedRecordEvent.java

```diff
@@ -51,7 +57,16 @@ public HasNextDelimitedRecordEvent(DelimitedRecordChannel channel, EventContext
      */
     @Override
     public EventResult get() {
-        boolean hasNext = channel.hasNext();
-        return new BooleanResult(hasNext, context);
+        BooleanResult result;
+        try {
+            boolean hasNext = channel.hasNext();
+            result = new BooleanResult(hasNext, context);
+        } catch (IOException e) {
+            String message = "Error occurred while reading bytes for hasNext()";
+            log.error(message, e);
+            context.setError(e);
```

Comment:
If we are returning the error to the user, I think we don't need to log it..

Reply:
As discussed with @SupunS since we provide the error details to the user it would not be mandatory to log the error. However, there will not be any harm in logging. This could be maintained as a sys log.